sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/rx/WebExceptionRetryPolicyE2ETests.java

```diff
@@ -0,0 +1,303 @@
+
+
+        try {
+            CosmosDiagnostics cosmosDiagnostics =
+                this.performDocumentOperation(cosmosAsyncContainer, operationType, newItem).block();
+            System.out.println("dataPlaneRequestHttpTimeout() Diagnostics " + " " + cosmosDiagnostics);
+            validateDataPlaneRetryPolicyResponseTimeouts(cosmosDiagnostics);
+        } catch (Exception e) {
+            fail("dataPlaneRequestHttpTimeout() should succeed for operationType " + operationType, e);
+        } finally {
+            requestHttpTimeoutRule.disable();
+        }
+    }
+
+    private void validateDataPlaneRetryPolicyResponseTimeouts(CosmosDiagnostics cosmosDiagnostics) {
+        List<ClientSideRequestStatistics.GatewayStatistics> gatewayStatisticsList = diagnosticsAccessor.getClientSideRequestStatistics(cosmosDiagnostics)
+            .stream()
+            .map(ClientSideRequestStatistics::getGatewayStatisticsList)
+            .flatMap(Collection::stream)
+            .collect(Collectors.toList());
+
+        for (ClientSideRequestStatistics.GatewayStatistics gs : gatewayStatisticsList) {
+            for (RequestTimeline.Event event : gs.getRequestTimeline()) {
+                Duration durationInMillis = event.getDuration();
+                if (durationInMillis != null) {
+                    assertThat(durationInMillis.getSeconds()).isLessThanOrEqualTo(62);
+                }
+            }
+        }
+    }
+
+    private void validateAddressRefreshRetryPolicyResponseTimeouts(CosmosDiagnostics cosmosDiagnostics) {
+
+        List<ClientSideRequestStatistics.AddressResolutionStatistics> addressResolutionStatisticsList = diagnosticsAccessor.getClientSideRequestStatistics(cosmosDiagnostics)
+            .stream()
+            .map(ClientSideRequestStatistics::getAddressResolutionStatistics)
+            .flatMap(m -> m.values().stream())
+            .sorted(Comparator.comparing(ClientSideRequestStatistics.AddressResolutionStatistics::getStartTimeUTC))
+            .collect(Collectors.toList());
+
+        assertThat(MILLIS.between(addressResolutionStatisticsList.get(0).getStartTimeUTC(), addressResolutionStatisticsList.get(0).getEndTimeUTC())).isLessThanOrEqualTo(600);
+        assertThat(MILLIS.between(addressResolutionStatisticsList.get(1).getStartTimeUTC(), addressResolutionStatisticsList.get(1).getEndTimeUTC())).isLessThanOrEqualTo(600);
+        assertThat(SECONDS.between(addressResolutionStatisticsList.get(2).getStartTimeUTC(), addressResolutionStatisticsList.get(2).getEndTimeUTC())).isLessThanOrEqualTo(6);
+        assertThat(SECONDS.between(addressResolutionStatisticsList.get(3).getStartTimeUTC(), addressResolutionStatisticsList.get(3).getEndTimeUTC())).isLessThanOrEqualTo(11);
+    }
+
+    private Mono<CosmosDiagnostics> performDocumentOperation(
+        CosmosAsyncContainer cosmosAsyncContainer,
+        OperationType operationType,
+        TestItem createdItem) {
+        if (operationType == OperationType.Query) {
```

Comment:
Instead of using these different if cases, why not use a switch case to make it simpler?

Reply:
Added switch case.